Sources/Overload/OvCore/src/OvCore/Helpers/Serializer.cpp

```diff
@@ -190,247 +184,398 @@ void OvCore::Helpers::Serializer::SerializeSound(tinyxml2::XMLDocument& p_doc, t
 	SerializeString(p_doc, p_node, p_name.c_str(), p_value ? p_value->path : "?");
 }
 
-void OvCore::Helpers::Serializer::DeserializeString(tinyxml2::XMLDocument & p_doc, tinyxml2::XMLNode * p_node, const std::string & p_name, std::string & p_out)
+void OvCore::Helpers::Serializer::DeserializeBoolean(tinyxml2::XMLDocument & p_doc, tinyxml2::XMLNode * p_node, const std::string & p_name, bool & p_out, bool p_default)
+{
+	if (auto element = p_node->FirstChildElement(p_name.c_str()); element)
+	{
+		tinyxml2::XMLError errorTest = element->QueryBoolText(&p_out);
+		if (errorTest != tinyxml2::XML_SUCCESS)
+		{
+			p_out = p_default;
+		}
+	}
+}
+
+void OvCore::Helpers::Serializer::DeserializeString(tinyxml2::XMLDocument & p_doc, tinyxml2::XMLNode * p_node, const std::string & p_name, std::string & p_out, std::string p_default)
 {
 	if (auto element = p_node->FirstChildElement(p_name.c_str()); element)
 	{
 		const char* result = element->GetText();
-		p_out = result ? result : "";
+		if (result == nullptr)
```

Comment:
In the case you want to work on that again in the future, here you do not need the statement since is there already the ternary operation that check the result, then you can just replace the `""` by the default value.